pkgs/tools/security/gnupg/22.nix

```diff
@@ -38,8 +38,10 @@ stdenv.mkDerivation rec {
     ./accept-subkeys-with-a-good-revocation-but-no-self-sig.patch
   ];
   postPatch = ''
-    sed -i 's,hkps://hkps.pool.sks-keyservers.net,hkps://keys.openpgp.org,g' \
-        configure doc/dirmngr.texi doc/gnupg.info-1
+    sed -i 's,hkps://hkps.pool.sks-keyservers.net,hkps://keys.openpgp.org,g' configure doc/dirmngr.texi doc/gnupg.info-1
+    # Fix broken SOURCE_DATE_EPOCH usage
+    sed -i 's/$SOURCE_DATE_EPOCH/''${SOURCE_DATE_EPOCH}/' doc/Makefile.am
+    sed -i 's/$SOURCE_DATE_EPOCH/''${SOURCE_DATE_EPOCH}/' doc/Makefile.in
```

Comment:
Why need to fix it here as well? Would not the `.in` file be generated from `.am`?

Reply:
Also would not patch like https://salsa.debian.org/debian/gnupg2/blob/debian/master/debian/patches/debian-packaging/avoid-regenerating-defsincdate-use-shipped-file.patch be nicer? It is easier to understand, and less likely to rot, IMHO.

Reply:
Or maybe https://github.com/gpg/gnupg/commit/61bb75d045a3709d1cba0084c95e991dfd52c8ee, since it appears to do the same you are attempting, if I understand automake syntax correctly.

Reply:
I attached the Debian patch in my other PR https://github.com/NixOS/nixpkgs/pull/93937, but then I realized we should probably use SOURCE_DATE_EPOCH since this is already supported (although) broken.

Reply:
I don't mind applying https://github.com/gpg/gnupg/commit/61bb75d045a3709d1cba0084c95e991dfd52c8ee and adding `autoreconfHook`, but I doubrt that is better. Hopefully we can remove the change on the next upstream release, so I vote keeping it as it is.

I added the note the `sed`s can be removed on the next upstream update in e296e89d75ab8aa1d0eed1c3688883a4f7937515

Reply:
Another thought: If we use `replace-literal -T` instead of `sed`to fix the URLs we can simply ignore the broken SOURCE_DATE_EPOCH, since that path in the Makefiles should never be triggered. :)

Reply:
I don't understand your `replace-literal` solution, feel free to elaborate. With this patch and version of gnupg (with your upstream sources, @wamserma) the build is entirely deterministic based solely on the hash of the upstream sources, since our modification in `postPatch` etc. no longer influences the `SOURCE_DATE_EPOCH` the build is modified to take into account.

Unless someone @jtojnar insists on autoreconf'ing this to avoid the `sed`'ing, I will merge this tomorrow. :)